Compatibility with Python `> 3.12`

Hi there, I'm seeing compatibility errors with the pinned versions resolving and building with Python 3.14 & Python 3.13. Using Python 3.12 works as stated.

# Python 3.14

```shell
➜  src git:(main) uv sync
Using CPython 3.14.0
Creating virtual environment at: .venv
Resolved 132 packages in 8ms
error: Distribution `spacy==3.8.14 @ registry+https://pypi.org/simple` can't be installed because it doesn't have a source distribution or wheel for the current platform

hint: You're using CPython 3.14 (`cp314`), but `spacy` (v3.8.14) only has wheels with the following Python ABI tags: `cp312`, `cp313`
```

# Python 3.13

```shell
Creating virtual environment at: .venv
warning: The workspace `requires-python` value (`==3.13`) contains an exact match without a patch version. When omitted, the patch version is implicitly `0` (e.g., `==3.13.0`). Did you mean `==3.13.*`?
Resolved 132 packages in 302ms
      Built docopt==0.6.2
      Built miniaudio==1.61
  × Failed to build `curated-tokenizers==0.0.9`
  ├─▶ The build backend returned an error
  ╰─▶ Call to `setuptools.build_meta.build_wheel` failed (exit status: 1)

      [stdout]
      Cythonizing sources
      running bdist_wheel
      running build
      running build_py
      creating build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/__init__.py -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      creating build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers/tests
      copying curated_tokenizers/tests/test_bbpe_processor.py -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers/tests
      copying curated_tokenizers/tests/test_word_piece_processor.py ->
      build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers/tests
      copying curated_tokenizers/tests/__init__.py -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers/tests
      copying curated_tokenizers/tests/test_sp.py -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers/tests
      running egg_info
      writing curated_tokenizers.egg-info/PKG-INFO
      writing dependency_links to curated_tokenizers.egg-info/dependency_links.txt
      writing requirements to curated_tokenizers.egg-info/requires.txt
      writing top-level names to curated_tokenizers.egg-info/top_level.txt
      reading manifest file 'curated_tokenizers.egg-info/SOURCES.txt'
      adding license file 'LICENSE'
      writing manifest file 'curated_tokenizers.egg-info/SOURCES.txt'
      copying curated_tokenizers/_bbpe.cpp -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/_bbpe.pxd -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/_bbpe.pyx -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/_spp.cpp -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/_spp.pxd -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/_spp.pyx -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/_wordpiece.cpp -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/_wordpiece.pxd -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/_wordpiece.pyx -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/config.h -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/merges.cc -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/merges.hh -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/py.typed -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/util.hh -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/wordpiece.cc -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/wordpiece.hh -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers
      copying curated_tokenizers/tests/incorrect-merges.txt -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers/tests
      copying curated_tokenizers/tests/robbert-merges-1000.txt ->
      build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers/tests
      copying curated_tokenizers/tests/robbert-vocab-1000.json ->
      build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers/tests
      copying curated_tokenizers/tests/toy-word-pieces.txt -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers/tests
      copying curated_tokenizers/tests/toy.model -> build/lib.macosx-11.0-arm64-cpython-313/curated_tokenizers/tests
      running build_ext
      building 'curated_tokenizers._spp' extension
      creating build/temp.macosx-11.0-arm64-cpython-313/curated_tokenizers
      creating build/temp.macosx-11.0-arm64-cpython-313/sentencepiece/src
      creating build/temp.macosx-11.0-arm64-cpython-313/sentencepiece/src/builtin_pb
      creating build/temp.macosx-11.0-arm64-cpython-313/sentencepiece/third_party/absl/flags
      creating build/temp.macosx-11.0-arm64-cpython-313/sentencepiece/third_party/absl/strings
      creating build/temp.macosx-11.0-arm64-cpython-313/sentencepiece/third_party/protobuf-lite
      c++ -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -g -O3 -Wall -arch arm64
      -mmacosx-version-min=11.0 -Wno-nullability-completeness -Wno-expansion-to-defined -Wno-undef-prefix -fPIC
      -Werror=unguarded-availability-new -Icurated_tokenizers -Isentencepiece -Isentencepiece/src -Isentencepiece/src/builtin_pb
      -Isentencepiece/third_party/protobuf-lite -I/Users/yossifrenkel/.cache/uv/builds-v0/.tmpztZf3Q/include
      -I/Users/yossifrenkel/.local/share/uv/python/cpython-3.13.0-macos-aarch64-none/include/python3.13 -c
      curated_tokenizers/_spp.cpp -o build/temp.macosx-11.0-arm64-cpython-313/curated_tokenizers/_spp.o --std=c++14
      -Wno-sign-compare-Wno-strict-prototypes -Wno-unused-function -D_USE_INTERNAL_STRING_VIEW -pthread -DHAVE_PTHREAD=1

      [stderr]
      ~/.cache/uv/builds-v0/.tmpztZf3Q/lib/python3.13/site-packages/setuptools/dist.py:765:
      SetuptoolsDeprecationWarning: License classifiers are deprecated.
      !!

              ********************************************************************************
              Please consider removing the following classifiers in favor of a SPDX license expression:

              License :: OSI Approved :: MIT License

              See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
              ********************************************************************************

      !!
        self._finalize_license_expression()
      warning: unknown warning option '-Wno-sign-compare-Wno-strict-prototypes' [-Wunknown-warning-option]
      curated_tokenizers/_spp.cpp:1198:10: fatal error: 'ios' file not found
       1198 | #include "ios"
            |          ^~~~~
      1 warning and 1 error generated.
      error: command '/usr/bin/c++' failed with exit code 1

      hint: This usually indicates a problem with the package or the build environment.
  help: `curated-tokenizers` (v0.0.9) was included because `parlor` (v0.1.0) depends on `misaki[en]` (v0.9.4) which depends on
        `spacy-curated-transformers` (v0.3.1) which depends on `curated-tokenizers`
```

--- a/src/pyproject.toml
+++ b/src/pyproject.toml
@@ -2,7 +2,7 @@
 name = "parlor"
 version = "0.1.0"
 description = "On-device, real-time multimodal AI — have natural voice + vision conversations, fully local and private"
-requires-python = ">=3.12"
+requires-python = ">=3.12,<3.13"
 dependencies = [
     "fastapi>=0.135.3",
     "kokoro-onnx>=0.5.0; sys_platform == 'linux'",

